src/main/java/org/mvel2/util/ParseTools.java

```diff
@@ -509,13 +514,16 @@ else if (arguments.length == 0 && parmTypes.length == 0) {
       }
 
       int score = getMethodScore(arguments, requireExact, parmTypes, isVarArgs);
+
       if (score != 0) {
         if (score > bestScore) {
           bestCandidate = construct;
           bestScore = score;
+          bestCandidateIsVarArgs = isVarArgs;
         }
-        else if (score == bestScore && (isMorePreciseForBigDecimal(construct, bestCandidate, arguments) || !isVarArgs)) {
+        else if (score == bestScore && (isMorePreciseForBigDecimal(construct, bestCandidate, arguments) || (bestCandidateIsVarArgs && !isVarArgs))) {
```

Comment:
Previously, `bestCandidate` was changed if it's not a varArgs constructor regardless of `isMorePreciseForBigDecimal` result.